### ISSUE TITLE
Android: registerAllFlagsListener is passing a stringified array instead of an array of strings to the given callback

### ISSUE THREAD

**Issue by rodperottoni** [launchdarkly/react-native-client-sdk#89]
**Is this a support request?**
No.

**Describe the bug**
On Android, the callback given to `registerAllFlagsListener` is receiving a stringified array of feature flags instead of _an actual `array`_ of strings. This is inconsistent with iOS & the typescript definitions provided by the library.

See images below:

**Type definition + example code:**
<img width="672" alt="Screen Shot 2021-09-03 at 10 36 09 am" src="https://user-images.githubusercontent.com/15217227/131933085-2f6d51e7-8da6-4341-a8b3-f405fd6dddd9.png">

**Console output:**
<img width="158" alt="Screen Shot 2021-09-03 at 10 36 20 am" src="https://user-images.githubusercontent.com/15217227/131933088-d9c2bdcc-8d84-4574-b708-2f301772b87f.png">

**To reproduce**
On a brand new app, initialise the LaunchDarkly SDK & use the following snippet:

```
client = new LaunchDarkly();
client.registerAllFlagsListener('ALL_FLAGS_LISTENER', (updatedFlags) => {
    console.log(typeof updatedFlags);
});
```

Then run the app on both iOS and Android and update a feature flag from the LaunchDarkly web app. The behaviour will differ between platforms.

**Expected behavior**
An array of strings representing which flags have changed should be passed in to the callback.

**Logs**
Non applicable.

**SDK version**

`"launchdarkly-react-native-client-sdk": "5.0.0"`

**Language version, developer tools**
Non applicable.

**OS/platform**
- MacOS Big Sur
- Android 9
- Tested on both React Native 0.65.1 & 0.64.2

**Comment by gwhelanLD** [comment]
Thanks for the issue report and associated (PR)[https://github.com/launchdarkly/react-native-client-sdk/pull/91]! I've already gone ahead and merged the PR, we'll have a patch release out when we've finished preparing an additional fix for this release.

Thanks!
@gwhelanLD

**Comment by gwhelanLD** [comment]
We've not released [5.0.1](https://github.com/launchdarkly/react-native-client-sdk/releases/tag/5.0.1), which includes your PR.

Thanks!
@gwhelanLD
